### PR TITLE
switch to background kudo manager

### PR DESCRIPTION
**Pull request by kensipe**
Signed-off-by: Ken Sipe <kensipe@gmail.com>

### CODE REVIEW

**Comment by kensipe** [comment]
to reviewers:   I'm not excited about the sleep 10 command (I'm expect to get comments on it :)... however I feel this is an ok compromise until we have a real solution which we should expect next week.

There is 1 more issue here that is worth mentioning... `manager` has a linux download option off our releases but does NOT have a darwin download option.   This means that local `make test` will fail on macs until the release process for kudo is updated (which is already on a PR for inclusion).   Monday I will also look into backporting the darwin release for the current version of kudo.

**Review by zmalik — APPROVED** [review]
LGTM

we will come back to remove the sleep 👍

File: Makefile [hunk 1]
```diff
@@ -26,14 +27,28 @@ bin/kubectl-kudo_$(KUDO_VERSION): bin/
...
+	curl -Lo bin/manager_$(KUDO_VERSION) https://github.com/kudobuilder/kudo/releases/download/v$(KUDO_VERSION)/manager_$(KUDO_VERSION)_$(OS)_$(KUDO_MACHINE)
+	chmod +x bin/manager_$(KUDO_VERSION)
+	ln -sf ./manager_$(KUDO_VERSION) ./bin/manager
+
+bin/kubectl-kuttl_$(KUTTL_VERSION): bin/
+	curl -Lo bin/kubectl-kuttl_$(KUTTL_VERSION) https://github.com/kudobuilder/kuttl/releases/download/v$(KUTTL_VERSION)/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)_$(KUDO_MACHINE)
+	chmod +x bin/kubectl-kuttl_$(KUTTL_VERSION)
+	ln -sf ./kubectl-kuttl_$(KUTTL_VERSION) ./bin/kubectl-kuttl
+
+install-kudo: bin/kubectl-kudo_$(KUDO_VERSION) bin/manager_$(KUDO_VERSION)
```
Comment (porridge) [inline Makefile:41]: This and the following should be `.PHONY`

**Review by porridge — APPROVED** [review]
One inline nitpick and a few questions:
- why do we need both `kudo-test.yaml` and `kuttl-test.yaml`?
- the sequencing of `kudo init`, `manger` and `sleep` makes it seem as if we are sleeping for the manager to start up, where in fact we are waiting for the CRDs to be processed. Can you please reorder the steps so that it's obvious? (I'm assuming the manager is resilient to CRDs not being in place for a short while, otherwise it will be flaky even with the current ordering.)

**Comment by kensipe** [comment]
@porridge thanks for your eagle eyes! and review.

The `kuttl-test.yaml` is what we are switching to... however we should have tests that are running kudo and need the `kudo-test.yaml`.   It is likely this is temporary until we have complete moved over.   I am not against keeping both.  It depends on if we want to continue to support `kudo test` function (which calls kuttl... we should also have kudo test recognize kuttl-test.yaml).   Regardless, this is temp needed.

Regarding the order of things... you are correct but I would rather not reorder things.  If the manager gets the advantage of loading during the sleep... that is preferred.  If we are testing resilents.. than we should have a specific test for that.. not an accident of test design (I'm do not mean to insinuated that you mean that... just thinking out loud).   The sleep is offensive to me and likely naive.  It is temporary.   We have several feature in mind to remove this all together.   The end result I hope will be more satisfactory to both of us.   (Also... temp regarding this is this week... I expect to work on it this week)

**Comment by porridge** [comment]
>  If the manager gets the advantage of loading during the sleep

FTR the idea was:
1. start manager
2. post CRDs
3. sleep

so that manager still has a chance to load during the sleep...

**Pull request merged**